iceoryx_hoofs/include/iceoryx_hoofs/cxx/string.hpp

```diff
@@ -357,55 +359,55 @@ class string
     /// the contents of both strings are equal, an integer > 0 if the first character that does not match has a greater
     /// value in self than in other
     template <typename T>
-    IsStringOrCharArray<T, int64_t> compare(const T& other) const noexcept;
+    auto compare(const T& other) const noexcept -> IsStringOrCharArray<T, int64_t>;
```

Comment:
@saif-at-github @FerdinandSpitzschnueffler Why is this notation preferred? It is equivalent but usually `auto` return type is used if the return type depends on compile time magic. In this case it cannot be stated after `->` explicitly but here it is additionally stated as well so the usefulness of `auto`is lost to me (it is just a more verbose way to express the same thing).

I suggest to not use this notation together with `auto` except in the case I stated above (where it is unavoidable).

Reply:
@saif-at-github I know, but this is a purely syntactic rule and I would argue we should not follow it.

Reply:
@FerdinandSpitzschnueffler @saif-at-github We should disable this rule after discussion if necessary.